### ISSUE TITLE
Metal Gear 2 - invisible boss

### ISSUE THREAD

**Issue by pjagiello** [openMSX/openMSX#1359]
... so, simply put, the eponymous machine of Metal Gear 2: Solid Snake is invisible for me on openMSX 16.0 on Linux. To reproduce that, you can load the first save from the attached disk and enter the fight. The machine I tried first was "Boosted MSX2+ JP" but it seems to happen no matter what machine I choose.

Save disk here:
[mg2.dsk.gz](https://github.com/openMSX/openMSX/files/6511771/mg2.dsk.gz)

Here's how it looks like:
![2021-05-20-00:19:02-screenshot](https://user-images.githubusercontent.com/8558288/118892369-1ac29e00-b901-11eb-9dd0-3fbef6b0b4a5.png)

A couple of comments:
- the save itself wasn't produced with the original game, but with the English fan translation... I would hope that doesn't matter though, unless something about it is so broken that it produces weird saves - the issue also occurs for me in the original, Japanese version of the game.
- I tested the same ROM and the same save on openMSX 16.0 on Windows... and oddly enough this doesn't happen here. (I also tested with fMSX on Linux, it also seems fine).
- I originally had this issue with "openmsx" package in Arch user repository, but then I downloaded the source and compiled openMSX myself just to be sure - the issue still happens.
- g++ version 11.1.0 was used for compilation.

Let me know if I can provide any more information.

**Comment by MBilderbeek** [comment]
Can you try with the latest source code from GitHub? I cannot reproduce the issue with the latest code on Linux.

**Comment by MBilderbeek** [comment]
I also can't reproduce it with openMSX 16.0 under Linux (Debian);
![3](https://user-images.githubusercontent.com/12675978/119056606-49a54680-b9cb-11eb-94eb-e4298ccd85e5.png)

**Comment by pjagiello** [comment]
Welp, same thing:
![2021-05-21-00:54:02-screenshot](https://user-images.githubusercontent.com/8558288/119058691-2381a580-b9cf-11eb-8f71-db2d61f408de.png)
... I would imagine the main difference with Debian setup would be GCC version, Buster seems to use 8.3.0 by default, and GCC11 came out just last month... maybe I'll try downgraded GCC next

**Comment by MBilderbeek** [comment]
I'm on Debian testing with g++-10.

Did you try to remove your settings.xml file? (Long shot ... But you never
know.)

Op vr 21 mei 2021 00:56 schreef Piotr Jagiełło ***@***.***>:

**Comment by pjagiello** [comment]
Well, what do you know, long shot worked!
![2021-05-21-01:04:49-screenshot](https://user-images.githubusercontent.com/8558288/119059441-993a4100-b9d0-11eb-91a8-b096816e0638.png)
... so did I mess up some setting inadvertently?

**Comment by MBilderbeek** [comment]
Apparently. Do you still have the file somewhere? Or do you remember which
settings you played with?

Perhaps the VDP cmdtiming setting? Or CPU speed?

Op vr 21 mei 2021 01:06 schreef Piotr Jagiełło ***@***.***>:

**Comment by pjagiello** [comment]
Well, I didn't have the foresight to save it before deleting just now. I don't think I played with any setting like that... certainly nothing that isn't available in Catapult.
At least not recently... the file was probably created with a version earlier than 0.16, maybe I did play around with something when I used it previously and I don't remember. But no obvious setting changed that I can find.

Well, I suppose the issue is resolved, since now I can't reproduce the invisible Metal Gear. Thanks for your help!
